src/main/java/rx/internal/operators/OperatorMerge.java

```diff
@@ -213,7 +213,7 @@ private void handleNewSource(Observable<? extends T> t) {
             InnerSubscriber<T> i = new InnerSubscriber<T>(this, producerIfNeeded);
             i.sindex = childrenSubscribers.add(i);
             t.unsafeSubscribe(i);
-            if (!isUnsubscribed()) {
+            if ((producerIfNeeded == null || producerIfNeeded.requested > 0) && !isUnsubscribed()) {
```

Comment:
@akarnokd the original code calls request(1) regardless of the "requested" variable causing the operator to potentially buffer a  dangerous amount of items and also a loop between producer.request() -> consumer.onNext() -> producer->request() ...